### PR TITLE
Custom sorter

### PR DESCRIPTION
**Pull request by ClemensSahs**
this PR ist for https://github.com/Siilwyn/css-declaration-sorter-atom/issues/49

- [x] add the ability to use the custom sorter
- [x] tests

### CODE REVIEW

**Comment by ClemensSahs** [comment]
until we have https://github.com/Siilwyn/css-declaration-sorter-atom/issues/110 tests are not realy usefull

File: lib/css-declaration-sorter.js [hunk 1]
```diff
@@ -42,26 +44,35 @@ module.exports = {
...
...sis-style, text-indent, text-justify, text-underline-position, text-orientation, text-overflow, text-rendering, text-shadow, text-size-adjust, text-transform, vertical-align, white-space, word-break, word-spacing, overflow-wrap, all, animation, animation-name, animation-duration, animation-fill-mode, animation-play-state, animation-timing-function, animation-delay, animation-iteration-count, animation-direction, mix-blend-mode, break-before, break-after, break-inside, page-break-before, page-break-after, page-break-inside, caption-side, clip-path, content, counter-increment, counter-reset, cursor, empty-cells, filter, image-orientation, image-rendering, mask, mask-clip, mask-image, mask-origin, mask-position, mask-repeat, mask-size, mask-type, opacity, perspective, perspective-origin, pointer-events, quotes, resize, scroll-behavior, scroll-snap-coordinate, scroll-snap-destination, scroll-snap-type, shape-image-threshold, shape-margin, shape-outside, tab-size, table-layout, text-combine-upright, touch-action, transition, transition-delay, transition-duration, transition-property, transition-timing-function, will-change, unicode-bidi, unicode-range, user-select, widows, writing-mod',
```
Comment (Siilwyn) [inline lib/css-declaration-sorter.js:66]: I don't think we need a default value for this?

**Comment by ClemensSahs** [comment]
will Require #111

File: lib/css-declaration-sorter.js (hunk 1 above)
Comment (ClemensSahs) [reply lib/css-declaration-sorter.js]: yes this is a valid point

**Comment by ClemensSahs** [comment]
in my mind we exclude the custom config file feature into a separat feature

File: lib/css-declaration-sorter.js [hunk 2]
```diff
@@ -42,18 +44,27 @@ module.exports = {
...
       default: 'alphabetical',
-      enum: ['alphabetical', 'smacss', 'concentric-css'],
+      enum: ['alphabetical', 'smacss', 'concentric-css', 'custom'],
     },
+    customOrder: {
+      order: 3,
+      title: 'Custom sorting order',
+      description: 'Select the default sorting order.',
+      type: 'string',
+      default: 'place your own order',
```
Comment (Siilwyn) [inline lib/css-declaration-sorter.js:66]:
```suggestion
      default: '',
```
Maybe just leave it empty? Or have a tiny example like `z-index border`

**Comment by Siilwyn** [comment]
I've opened PR #114, it's your changes but rebased on master so there are no conflicts.

**Pull request closed without merging**